### PR TITLE
docs(api-gateway): add support for new router feature 

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

## Description of changes:

Add docs on how to use the new `Router` decorator for the API Gateway Resolvers

**Checklist**

 

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

-----
[View rendered docs/core/event_handler/api_gateway.md](https://github.com/gyft/aws-lambda-powertools-python/blob/docs/router/docs/core/event_handler/api_gateway.md)

### CODE REVIEW

File: docs/core/event_handler/api_gateway.md [hunk 1]
```diff
@@ -853,6 +853,165 @@ You can instruct API Gateway handler to use a custom serializer to best suit you
...
+                    HealthPath:
+                        Type: Api
+                        Properties:
+                            Path: /health/status
+                            Method: GET
+                    ItemsPath:
+                        Type: Api
+                        Properties:
+                            Path: /items
+                            Method: GET
```
Comment (heitorlessa) [inline docs/core/event_handler/api_gateway.md:934]: Can we keep /users and /health only? 

That would be great to show the feature without distracting them from design questions like: "Shouldn't this be separately, etc. etc."

File: docs/core/event_handler/api_gateway.md [hunk 2]
```diff
@@ -853,6 +853,158 @@ You can instruct API Gateway handler to use a custom serializer to best suit you
         }
     ```
 
+### Splitting routes across multiple files
+
+When building a larger application, sometimes to helps to split out your routes into multiple file. Also
```
Comment (DanyC97) [inline docs/core/event_handler/api_gateway.md:858]:
```suggestion
When building a larger application, sometimes it helps to split out your routes into multiple file. Also
```

File: docs/core/event_handler/api_gateway.md [hunk 3]
```diff
@@ -853,6 +853,264 @@ You can instruct API Gateway handler to use a custom serializer to best suit you
...
+    ```python hl_lines="10-13"
+    .
+    ├── Pipfile					  # project dev dependencies
+    ├── Pipfile.lock
+    ├── src
+    │   ├── __init__.py
+    │   ├── requirements.txt      # dummy for `sam build`, as external deps are Lambda Layers
+    │   └── app
+    │       ├── __init__.py       # this file makes "app" a "Python package"
+    │       ├── main.py           # Main lambda handler (app.py, index.py, handler.py)
```
Comment (michaelbrewer) [inline docs/core/event_handler/api_gateway.md:981]: @heitorlessa -  Note: `main.py` was chosen as the `app` folder was taken.

File: docs/core/event_handler/api_gateway.md [hunk 4]
```diff
@@ -853,6 +853,264 @@ You can instruct API Gateway handler to use a custom serializer to best suit you
...
+=== "Project layout"
+
+    ```python hl_lines="10-13"
+    .
+    ├── Pipfile					  # project dev dependencies
+    ├── Pipfile.lock
+    ├── src
+    │   ├── __init__.py
+    │   ├── requirements.txt      # dummy for `sam build`, as external deps are Lambda Layers
+    │   └── app
```
Comment (michaelbrewer) [inline docs/core/event_handler/api_gateway.md:979]: @heitorlessa - we could rename this to the intent of the lambda itself like `users`

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**